airflow/contrib/operators/livy_operator.py

```diff
@@ -0,0 +1,175 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This module contains the Apache Livy operator.
+"""
+
+from time import sleep, gmtime, mktime
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException
+from airflow.contrib.hooks.livy_hook import LivyHook, BatchState, TERMINAL_STATES
+
+
+class LivyOperator(BaseOperator):
+    """
+    :param file: Path of the  file containing the application to execute (required).
+    :type file: str
+    :param class_name: Application Java/Spark main class string.
+    :type class_name: str
+    :param args: Command line arguments for the application s.
+    :type args: list
+    :param jars: jars to be used in this sessions.
+    :type jars: list
+    :param py_files: Python files to be used in this session.
+    :type py_files: list
+    :param files: files to be used in this session.
+    :type files: list
+    :param driver_memory: Amount of memory to use for the driver process  string.
+    :type driver_memory: str
+    :param driver_cores: Number of cores to use for the driver process int.
+    :type driver_cores: str
+    :param executor_memory: Amount of memory to use per executor process  string.
+    :type executor_memory: str
+    :param executor_cores: Number of cores to use for each executor  int.
+    :type executor_cores: str
+    :param num_executors: Number of executors to launch for this session  int.
+    :type num_executors: str
+    :param archives: Archives to be used in this session.
+    :type archives: list
+    :param queue: The name of the YARN queue to which submitted string.
+    :type queue: str
+    :param name: The name of this session  string.
+    :type name: str
+    :param conf: Spark configuration properties.
+    :type conf: dict
+    :param proxy_user: User to impersonate when running the job.
+    :type proxy_user: str
+    :param livy_conn_id: reference to a pre-defined Livy Connection.
+    :type livy_conn_id: str
+    :param polling_interval: time in seconds between polling for job completion. Don't poll for values >=0
+    :type polling_interval: int
+    :param timeout: for a value greater than zero, number of seconds to poll before killing the batch.
+    :type timeout: int
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        file=None,
+        args=None,
+        conf=None,
+        livy_conn_id='livy_default',
+        polling_interval=0,
+        timeout=24 * 3600,
+        *vargs,
```

Comment:
Why did you call them `*vargs` instead of `*args` ? Your documentation has only `*args` and I think `*args` is also more commonly used.

Reply:
`args` is also the name of one of the Livy arguments that goes in the body of the request.
What I'm trying to do here is maintain the Livy naming conventions for a better ease for the Spark user, and that meant the Python `*args` had to be renamed.
But since `*args` and `**kwargs` will go away as [per previous suggestion](https://github.com/apache/airflow/pull/6090#discussion_r326871236), this issue is no more.